app/models/ncr/work_order.rb

```diff
@@ -114,13 +114,23 @@ def approving_official
       self.approvers.first
     end
 
-    def current_approver_email_address
+    def current_approver
       if self.pending?
-        self.currently_awaiting_approvers.first.email_address
+        self.currently_awaiting_approvers.first
       elsif self.approving_official
-        self.approving_official.email_address
+        self.approving_official
+      elsif self.emergency and self.approvers.empty?
+        nil
       else
-        self.system_approver_emails.first
+        User.for_email(self.system_approver_emails.first)
+      end
+    end
+
+    def final_approver
+      if self.emergency and self.approvers.empty?
```

Comment:
AH yes, I see. I think the additional color you provided above gives more evidence that this could be more clearly constructed, because my first reading of what was happening wasn't correct. what about:

```ruby 
def final_approver
  if not_emergency? && approvers.any?
    self.approvers.last
  end
end
```